psi4/src/psi4/libsapt_solver/fdds_disp.cc

```diff
@@ -481,5 +542,547 @@ SharedMatrix FDDS_Dispersion::form_unc_amplitude(std::string monomer, double ome
 
     return ret;
 }
+
+std::vector<SharedMatrix> FDDS_Dispersion::form_aux_matrices(std::string monomer, double omega){
+
+    // => Configuration <= //
+    SharedVector eps_occ, eps_vir;
+    std::string arQ_name, XarQ_name, YarQ_name, QXarQ_name, QYarQ_name;
+
+    if (monomer == "A") {
+        eps_occ = vector_cache_["eps_occ_A"];
+        eps_vir = vector_cache_["eps_vir_A"];
+        arQ_name = "arQ";
+        XarQ_name = "XarQ";
+        QXarQ_name = "QXarQ";
+        YarQ_name = "YarQ";
+        QYarQ_name = "QYarQ";
+    } else if (monomer == "B") {
+        eps_occ = vector_cache_["eps_occ_B"];
+        eps_vir = vector_cache_["eps_vir_B"];
+        arQ_name = "bsQ";
+        XarQ_name = "XbsQ";
+        QXarQ_name = "QXbsQ";
+        YarQ_name = "YbsQ";
+        QYarQ_name = "QYbsQ";
+    } else {
+        throw PSIEXCEPTION("FDDS_Dispersion::form_aux_matrices: Monomer must be A or B!");
+    }
+
+
+    // => Sizing <= //
+
+    size_t nocc = eps_occ->dim(0);
+    size_t nvir = eps_vir->dim(0);
+    size_t naux = auxiliary_->nbf();
+    size_t nov = nocc * nvir;
+
+    // => Blocking <= //
+
+    size_t doubles = Process::environment.get_memory() * 0.8 / sizeof(double);
+    long long int rem = doubles - 2 * nocc * nvir - 6 * naux * naux;
+    if (rem < 0)
+        throw PSIEXCEPTION("Too little static memory for FDDS_Dispersion::form_aux_matrices()");
+
+    size_t maxo = rem / (7 * nvir * naux);
+    maxo = (maxo > nocc ? nocc : maxo);
+    if (maxo < 1)
+        throw PSIEXCEPTION("Too little static memory for FDDS_Dispersion::form_aux_matrices()");
+
+    // => Scalars <= //
+
+    auto Lar = std::make_shared<Matrix>(nocc, nvir); // lambda
+    auto LDar = std::make_shared<Matrix>(nocc, nvir); // lambda * d
+
+    double** Larp = Lar->pointer();
+    double** LDarp = LDar->pointer();
+    double* eoccp = eps_occ->pointer();
+    double* evirp = eps_vir->pointer();
+
+#pragma omp parallel for
+    for (size_t a = 0; a < nocc; a++) {
+        for (size_t r = 0; r < nvir; r++) {
+            double val = evirp[r] - eoccp[a]; // d
+            double ll = -4.0 / (val * val + omega * omega); // lambda
+            double ld = val * ll; // lambda * d
+            Larp[a][r] = ll;
+            LDarp[a][r] = ld;
+        }
+    }
+
+    // => Tensor Slices <= //
+
+    auto arQ = std::make_shared<Matrix>("arQ", maxo * nvir, naux);
+    auto arQLD = std::make_shared<Matrix>("arQLD", maxo * nvir, naux);
+    auto XarQ = std::make_shared<Matrix>("XarQ", maxo * nvir, naux);
+    auto QXarQ = std::make_shared<Matrix>("QXarQ", maxo * nvir, naux);
+    auto YarQ = std::make_shared<Matrix>("YarQ", maxo * nvir, naux);
+    auto QYarQ = std::make_shared<Matrix>("QYarQ", maxo * nvir, naux);
+    auto YarQL = std::make_shared<Matrix>("YarQL", maxo * nvir, naux);
+
+    // => Target <= //    
+    std::vector<SharedMatrix> ret;
+    for (size_t i = 0; i < 5; i++) {
```

Comment:
Instead of making a vector, I would generally recommend making additional matrices for clarity.